Remove SwapDirection

lubkoll opened the pull request.

magiodev commented on smart-contracts/osmosis/contracts/cl-vault/src/vault/swap.rs:
```diff
@@ -105,40 +99,30 @@ pub fn execute_swap_non_vault_funds(
     pool_config: PoolConfig,
-    swap_direction: SwapDirection,
-    token_in_amount: Uint128,
+    offer: Coin,
```
Is a bit confusing the way we call tokens, which sometimes is token_in and here is offer.

lubkoll replied:
done

magiodev commented on smart-contracts/osmosis/contracts/cl-vault/src/vault/range.rs:
```diff
@@ -262,61 +262,77 @@ fn do_swap_deposit_merge(
         pool_config,
-        swap_direction,
-        token_in_amount,
+        token_in,
```
Here is token_in

lubkoll replied:
done

magiodev reviewed (approved):
Mostly looks good, just a minor comment.

LaurensKubat commented on smart-contracts/osmosis/contracts/cl-vault/src/contract.rs:
```diff
@@ -292,8 +290,6 @@ pub fn migrate(deps: DepsMut, _env: Env, msg: MigrateMsg) -> Result<Response, Co
     #[allow(deprecated)]
     CURRENT_BALANCE.remove(deps.storage);
-    #[allow(deprecated)]
-    CURRENT_SWAP.remove(deps.storage);
```
Afaik this remove was not migrated yet to mainnet contracts, if so we should inline the remove so we don't orphan state items

lubkoll replied:
I don't get what you refer to with "orphaning" items, what is that and what is the real problem with removing this?

LaurensKubat commented on smart-contracts/osmosis/contracts/cl-vault/src/vault/deposit.rs:
```diff
@@ -78,57 +75,70 @@ pub(crate) fn execute_any_deposit(
-        let swap_amount = if pool_details.current_tick > position.upper_tick {
-            deposit_info.base_refund
+    let twap_price = get_twap_price(deps.storage, &deps.querier, &env, 24u64)?;
+    let (offer, out_denom, remainder, price) = if !deposit_info.base_refund.is_zero() {
```
If we change offer according to Stefano's comments, lets change it here to so we have token_in, out_denom

lubkoll replied:
done

LaurensKubat reviewed (commented):
generally LGTM, 2 small comments

lubkoll commented:
I removed some failing code in a test-tube test (in authz::exact_deposit_withdraw_equal).
Happy to add it back if somebody is able to explain what is tested there and why we need this test.

lubkoll commented:
> I removed some failing code in a test-tube test (in authz::exact_deposit_withdraw_equal). Happy to add it back if somebody is able to explain what is tested there and why we need this test.

Added the test back..

ajansari95 reviewed (approved):
lgtm
